### PR TITLE
feat(DocSearch): add poc link under Useful Links

### LINKED ISSUE #5120: Request a POC - Useful Links
ACs
    - Add an option to the Useful Links list called "Request a Proof of Concept"
    - If a user clicks "Request a Proof of Concept" in the Useful Links section, send them here: https://www.influxdata.com/proof-of-concept/
    - Add eventing to track when users click the 'request a POC' option via the Useful Links
    - Manage with a feature flag
    - Keep behind a feature flag until PM or Engineering Management gives the green light to flip on in prod
    - Keep the feature flag available since this is a growth campaign

### PR DESCRIPTION
**Pull request by blegesse-git**
Closes #5120 

Adds `Request Proof of Concept` under Useful Links of UI homepage. 
behind `requestpoc` feature flag. 
Screen Shot 2022-07-29 at 11 56 12 AM

### Checklist

Authors and Reviewer(s), please verify the following:

- [x] A PR description, regardless of the triviality of this change, that communicates the value of this PR
- [x] [Well-formatted conventional commit messages](https://www.conventionalcommits.org/en/v1.0.0/) that provide context into the change
- [x] Documentation updated or issue created (provide link to issue/PR)
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [x] Feature flagged, if applicable

### CODE REVIEW

File: src/me/components/DocSearchWidget.tsx [hunk 1]
```diff
@@ -34,8 +40,18 @@ const supportLinks = [
...
+    title: 'Request Proof of Concept',
+  },
 ]
 
+if (CLOUD && !isFlagEnabled('requestPoc')) {
+  supportLinks = supportLinks.filter(
+    item => item.title !== 'Request Proof of Concept'
+  )
+}
+
```
Comment (hoorayimhelping) [inline src/me/components/DocSearchWidget.tsx:54]: I think it might be a little bit clearer to remove the `Request Proof of Concept` object from the `supportLinks` array and then do something like this:

```ts
if (CLOUD && !isFlagEnabled('requestPoc')) {
  supportLinks.push({
    link: 'https://www.influxdata.com/proof-of-concept/',
    title: 'Request Proof of Concept',
  })
}
```

What do you think? It feels a little more natural to me to add something under certain conditions rather than adding something then immediately removing them under certain conditions.

**Review by hoorayimhelping — COMMENTED** [review]
Thoughts on this suggestion?

File: src/me/components/DocSearchWidget.tsx (hunk 1 above)
Comment (blegesse-git) [reply src/me/components/DocSearchWidget.tsx:54]: yeah, that's a fair point. I think I opted out for filtering out poc object after it's added to make it easier for clean up. Whenever we decide to remove the `requestPoc` feature flag, all we will have to do is remove line 49-53 and leave the `supportLinks` object as is.

Comment (blegesse-git) [reply src/me/components/DocSearchWidget.tsx:54]: But considering how small of a task this is in general, let's go with your suggestion.

Comment (hoorayimhelping) [reply src/me/components/DocSearchWidget.tsx]: >I think I opted out for filtering out poc object after it's added to make it easier for clean up. Whenever we decide to remove the `requestPoc` feature flag, all we will have to do is remove line 49-53 and leave the `supportLinks` object as is. 

That's fair reasoning. And I know you said let's go with my suggestion, but I feel an explanation still might help. I think this falls under YAGNI - You Ain't Gonna Need It. We should write the code for the problem we're solving now, not the one in the future. In general, we'll know more in the future than we know now, and things in the future will change - "always in motion is the future". Writing code to solve future problems that may never arrive results in code that can end up confusing and hard to follow. It can result in code that tends to give the impression that the author knew something in the past more than we know now, which makes the code scary to change. Now like you said, this is a small change, but I think the reasoning is applicable to a lot of code that we write :)

**Review by hoorayimhelping — APPROVED** [review]

File: src/me/components/DocSearchWidget.tsx (hunk 1 above)
Comment (blegesse-git) [reply src/me/components/DocSearchWidget.tsx]: wow, that's definitely something I didn't consider. To summarize - writing code with preconceived notions can potentially lead to unexpected problems. Makes perfect sense.
